idunn/utils/result_filter.py

```diff
@@ -0,0 +1,183 @@
+import logging
+import re
+from typing import List
+from unidecode import unidecode
+
+from jellyfish import damerau_levenshtein_distance
+
+
+logger = logging.getLogger(__name__)
+
+
+# Typical suffixes found after numbers such as "4bis", "4th", ...
+NUM_SUFFIXES = ["bis", "ter", "quad", "e", "è", "eme", "ème", "er", "st", "nd", "rd", "th"]
+
+# Abrevation typical found in places, addresses, ...
+ABREVIATIONS = {
+    "av": "avenue",
+    "bd": "boulevard",
+    "r": "rue",
+    "st": "saint",
+}
+
+
+def words(text):
+    """
+    Split words into a list.
+
+    >>> words("17, rue jaune ; Levallois-Peret")
+    ['17', 'rue', 'jaune', 'Levallois', 'Peret']
+    """
+    separators = [" ", "-", ",", ";", "\\.", "'"]
+    return [word for word in re.split("|".join(separators), text) if word]
```

Comment:
There are many redundant calls to `.lower()` in this file. Maybe we could return lowercase words here ?